pkg/volume/nfs/nfs_test.go

```diff
@@ -181,6 +181,22 @@ func TestPluginVolume(t *testing.T) {
 	doTestPlugin(t, volume.NewSpecFromVolume(vol))
 }
 
+func TestIPV6VolumeSource(t *testing.T) {
+	vol := &v1.Volume{
+		Name:         "vol1",
+		VolumeSource: v1.VolumeSource{NFS: &v1.NFSVolumeSource{Server: "0:0:0:0:0:0:0:1", Path: "/somepath", ReadOnly: false}},
+	}
+	doTestPlugin(t, volume.NewSpecFromVolume(vol))
```

Comment:
doTestPlugin does not test that the plugin added `[ ]` around the IP address, you need to add it there.

Reply:
fixed 👍🏽

Reply:
@jsafrane `nfsMounter` does not expose `server` field, so that I can assert upon it. 

I have been trying to find an interface implemented by `nfsMounter` that exposes the field, but I could not.  I can :- 

```
 func (nfsMounter *nfsMounter) getServer() string {
	return nfsMounter.server
}
```
What do u think ?

Reply:
I meant something like this:
```patch
--- a/pkg/volume/nfs/nfs_test.go
+++ b/pkg/volume/nfs/nfs_test.go
@@ -96,7 +96,7 @@ func TestRecycler(t *testing.T) {
        }
 }
 
-func doTestPlugin(t *testing.T, spec *volume.Spec) {
+func doTestPlugin(t *testing.T, spec *volume.Spec, expectedDevice string) {
        tmpDir, err := utiltesting.MkTmpdir("nfs_test")
        if err != nil {
                t.Fatalf("error creating temp dir: %v", err)
@@ -146,9 +146,6 @@ func doTestPlugin(t *testing.T, spec *volume.Spec) {
                if mntDevs[0].Type != "nfs" {
                        t.Errorf("unexpected type of mounted devices. expected: %v, got %v", "nfs", mntDevs[0].Type)
                }
-               src, _, _ := getVolumeSource(spec)
-               srvr := getServerFromSource(src)
-               expectedDevice := fmt.Sprintf("%s:%s", srvr, src.Path)
                if mntDevs[0].Device != expectedDevice {
                        t.Errorf("unexpected nfs device, expected %q, got: %q", expectedDevice, mntDevs[0].Device)
                }
@@ -203,7 +200,7 @@ func TestIPV6VolumeSource(t *testing.T) {
                Name:         "vol1",
                VolumeSource: v1.VolumeSource{NFS: &v1.NFSVolumeSource{Server: "0:0:0:0:0:0:0:1", Path: "/somepath", ReadOnly: false}},
        }
-       doTestPlugin(t, volume.NewSpecFromVolume(vol))
+       doTestPlugin(t, volume.NewSpecFromVolume(vol), "[0:0:0:0:0:0:0:1]:/somepath")
 }
 
 func TestIPV4VolumeSource(t *testing.T) {
```

(+ you need to update all other `doTestPlugin` calls)
This way you can explicitly specify that `[]` should be added to IPv6 addresses and not to ipv4.

Reply:
@jsafrane added in the last commit 👍🏽